### ISSUE TITLE
rtr2: can't pass rtr2_settings or rtr2_controller_settings

### ISSUE THREAD

**Issue by ThomasCr** [smarthomeNG/plugins#647]
Hi,

I cant pass the item settings for `rtr2_settings` and `rtr2_controller_settings` to the plugin.

And for me this looks really untested:
https://github.com/smarthomeNG/plugins/blob/c16df49c7136c52bcbba360897384b6d84126b9a/rtr2/__init__.py#L165

1. why from the parent_item?

https://github.com/smarthomeNG/plugins/blob/c16df49c7136c52bcbba360897384b6d84126b9a/rtr2/__init__.py#L166

2. `rtr2_controller_settings` is not even evaluated

https://github.com/smarthomeNG/plugins/blob/c16df49c7136c52bcbba360897384b6d84126b9a/rtr2/__init__.py#L417

~~3. the value from `temp_settings` is never used - only the defaults get assigned~~

Maybe I am to stupid to understand the code, but there is also no example documentation.

Thats my test yaml:

```
        rtr:
            name: Badezimmer Oben
            sv_widget: "{{ rtr2.rtr('item') }}"
            sv_item_type: heater
            struct: rtr2.rtr
            rtr2_id: bz_og
            rtr2_settings:
              - 24
            rtr2_controller_settings:
              - 4.0
              - 60.0
            visu_acl: rw

            soll_temp:
                visu_acl: ro

            ist_temp:
                type: num
                eval: sh.og.bz.temperature()
                eval_trigger: og.bz.temperature
                database: true

            stellwert:
                type: num
                knx_dpt: '5.001'
                knx_send: 4/1/220
                knx_init: 4/1/220
                knx_listen: 4/1/220
                enforce_updates: true
                database: true

            nacht:
                eval: True if sh.zentral.heizung.nachtabsenkung() else False
                eval_trigger: zentral.heizung.nachtabsenkung

            frost:
                eval: True if sh.og.bz.fenster.status_rtr() else False
                eval_trigger: og.bz.fenster.status_rtr
```

**Comment by msinn** [comment]
I use the rtr2 plugin since about two year for heating my flat.

Have you had a look at the quite extensive documentation of the plugin (https://www.smarthomeng.de/user/plugins/rtr2/user_doc.html)?

**Comment by ThomasCr** [comment]
yes for sure - have u also taken a look at it? There is no word about the `rtr2_settings` or `rtr2_controller_settings`.
But at the following docs it is mentioned, but no example or closer description:
https://smarthomeng.github.io/smarthome/plugins_doc/config/rtr2.html#rtr2-settings
https://smarthomeng.github.io/smarthome/plugins_doc/config/rtr2.html#rtr2-controller-settings

Have you taken a closer look at the code part I linked up?

**Comment by msinn** [comment]
No I haben‘t looked ad the code and have no time to do so at the moment.

I can only say that the plugin is functional, I use it for years and it is extemsivly documented how to use the plugin. The easiest way to use the plugin is to use the struct which comes with the plugin.

**Comment by ThomasCr** [comment]
yes, but I need to tune my controller settings for a specific room. The defaults dont match there really well!

**Comment by ThomasCr** [comment]
I fixed the code, so that I can provide different values for Ki / Kp and also added documentation for the item params.

I dont really understand, why we need to get the `item.return_parent()` at this point, but it works.

**Comment by msinn** [comment]
Issue is fixed in plugin version v2.1.1
